docs/data/routes/docs/nextjs/creating-components/layout-concepts/en.md

```diff
@@ -0,0 +1,125 @@
+
+The logic for generating component factory is in [`scripts/generate-component-factory.ts`](https://github.com/Sitecore/jss/blob/dev/samples/nextjs/scripts/generate-component-factory.ts). By default, it recursively loops through `src/components` and all sub-directories, and processes all `.tsx` files, with the exception of `.d.tsx` (type definition) files.
+
+It does not matter whether components live directory under `src/components` or whether they're organized in folders. Both of the following src structures will generate a component factory with two components, named "SomeComponent" and "AnotherComponent".
+
+Example 1 - components directory under `src/components`
+```
+/src
+  /components
+    /SomeComponent.tsx
+    /AnotherComponent.tsx
+```
+
+Example 2 - components organized in folders
+```
+/src
+  /components
+    /styleguide
+      /SomeComponent.tsx
+      /AnotherComponent.tsx
+    /graphql
+      /SomeComponent.tsx
+      /AnotherComponent.tsx
+```
+
+This file structure aligns with how the JSS CLI generates new components. If you prefer a different structure, you can customize `scripts/generate-component-factory.ts`.
+
+Programmatic generation of component factory is not required; it just makes development easier by eliminating the need to manually register new components. Additionally, when running the app on a local web server, it is smart enough to watch component factory for changes, so new components can be added quickly. If you prefer to maintain the component factory manually, do the following:
+- Make sure `src/temp/componentFactory.ts` is included in source control
+- Delete `scripts/generate-component-factory.ts` and `scripts/templates/component-factory.ts`
+- Remove the reference to the deleted script from 'scripts/bootstrap.ts'
+
+## Creating Components
+
+### Using the CLI
+> Note: If you've worked with other JSS SDKs before (React, Angular, or Vue), note that the default component structure in the Next.js sample app is different from the other samples. We decided to make this change to align with the structure that Next.js authors use in their [code samples](https://github.com/vercel/next.js/tree/master/examples).
+
+New components can be added using the `jss scaffold <componentName>` command. The `componentName` parameter can be a component name or a relative path (relative to `src/components`). Examples:
+
+Command `jss scaffold ComponentName` will generate:
+- `src/components/ComponentName.tsx`
+- `sitecore/definitions/components/ComponentName.sitecore.tsx`
+
+Command `jss scaffold some/new/path/ComponentName` will generate files in the specified folder structure, creating new folders as needed:
+- `src/components/some/new/path/ComponentName.tsx`
+- `sitecore/definitions/components/some/new/path/ComponentName.sitecore.tsx`.
+
+If you don’t like the project structure generated by `jss scaffold`, you can customize `scripts/scaffold-component.ts`.
```

Comment:
Should we note specifically that they may want to remove the definition creation for Sitecore-first?

Reply:
Done